lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -61,19 +61,22 @@ def __init__(self, paths=None, scan_sys_paths=True):
         # expand any placeholders in configured paths
         paths = [os.path.expanduser(to_native(p, errors='surrogate_or_strict')) for p in paths]
 
+        # add syspaths if needed
         if scan_sys_paths:
-            # append all sys.path entries with an ansible_collections package
-            for path in sys.path:
-                if (
-                        path not in paths and
-                        os.path.isdir(to_bytes(
-                            os.path.join(path, 'ansible_collections'),
-                            errors='surrogate_or_strict',
-                        ))
-                ):
-                    paths.append(path)
-
-        self._n_configured_paths = paths
+            paths.extend(sys.path)
+
+        good_paths = []
+        # expand any placeholders in configured paths
+        for p in paths:
+
+            # ensure we alway shave ansible_collections
+            if os.path.basename(p) == 'ansible_collections':
+                p = os.path.dirname(p)
+
+            if p not in good_paths and os.path.isdir(to_bytes(os.path.join(p, 'ansible_collections'), errors='surrogate_or_strict')):
```

Comment:
There was a [unit test ](https://github.com/ansible/ansible/blob/devel/test/units/utils/collection_loader/test_collection_loader.py#L31) that was relying on the side effect that only system paths needed to be a dir, not the explicitly provided collection paths.

Reply:
@nitzmahone any specific reason why 'some paths' were verified and others not?

Reply:
IIRC this behavioral disconnect was because the two features were added at different times. My original assumption was that the collections path would remain a relatively short list of dirs, so allowing the normal Python behavior of "stuff can appear later and it's fine" to continue was OK, but when we started putting `sys.path` into the mix, I wanted to scan it only once, because it's not uncommon for `sys.path` to have hundreds of generated entries in it, and I didn't want the collection loader having to scan all that all the time. Off the top of my head, I don't really see a problem moving the whole thing to "if there's no ansible_collections package in it when we fire it up, just ignore it".